tests/cross/src/test/scala/tests/tokens/SemanticTokensSuite.scala

```diff
@@ -263,4 +265,31 @@ class SemanticTokensSuite extends BaseSemanticTokensSuite {
         |}""".stripMargin,
   )
 
+  check(
+    "pattern-match",
+    s"""|package <<example>>/*namespace*/
+        |
+        |class <<Imports>>/*class*/ {
+        |  
+        |  val <<a>>/*variable,definition,readonly*/ = <<Option>>/*class*/(<<Option>>/*class*/(""))
+        |  <<a>>/*variable,readonly*/ match {
+        |    case <<Some>>/*class*/(<<Some>>/*class*/(<<b>>/*variable,definition,readonly*/)) => <<b>>/*variable,readonly*/
+        |    case <<Some>>/*class*/(<<b>>/*variable,definition,readonly*/) => <<b>>/*variable,readonly*/
+        |    case <<other>>/*variable,definition,readonly*/ => 
+        |  }
+        |}""".stripMargin,
+  )
+
+  check(
+    "pattern-match-value",
+    s"""|package <<example>>/*namespace*/
+        |
+        |object <<A>>/*class*/ {
+        |  val <<x>>/*variable,definition,readonly*/ = <<List>>/*variable,readonly*/(1,2,3)
```

Comment:
Probably we have to add `compat` here to fix this test (`List` is `class` in 2.12.14